lib/pbench/server/cache_manager.py

```diff
@@ -1133,27 +1134,46 @@ def get_results(self, lock: LockManager) -> Path:
                 uend - ustart,
             )
 
-        self.last_ref.touch(exist_ok=True)
-
-        # If we have a Dataset, and haven't already done this, compute the
-        # unpacked size and record it in metadata so we can use it later.
-        ssize = time.time()
-        if self.dataset and not Metadata.getvalue(
-            self.dataset, Metadata.SERVER_UNPACKED
-        ):
-            try:
-                process = subprocess.run(
-                    ["du", "-s", "-B1", str(self.unpacked)],
-                    capture_output=True,
-                    text=True,
+            # If we have a Dataset, and haven't already done this, compute the
+            # unpacked size and record it in metadata so we can use it later.
+            ssize = time.time()
+            if self.dataset:
+                if not Metadata.getvalue(self.dataset, Metadata.SERVER_UNPACKED):
+                    try:
+                        process = subprocess.run(
+                            ["du", "-s", "-B1", str(self.unpacked)],
+                            capture_output=True,
+                            text=True,
+                        )
+                        if process.returncode == 0:
+                            size = int(process.stdout.split("\t", maxsplit=1)[0])
+                            self.unpacked_size = size
+                            Metadata.setvalue(
+                                self.dataset, Metadata.SERVER_UNPACKED, size
+                            )
+                    except Exception as e:
+                        self.logger.warning("usage check failed: {}", e)
+
+                # Update the time-to-unpack statistic. If the metadata doesn't exist,
+                # or somehow isn't a dict (JSONOBJECT), create a new metric: otherwise
+                # update the existing metric.
+                unpack_time = uend - ustart
+                metric: JSONOBJECT = Metadata.getvalue(
+                    self.dataset, Metadata.SERVER_UNPACK_PERF
                 )
```

Comment:
We also called `getvalue()` at line 1141.  Would it make sense to try to cache and reuse the value from that call instead of calling again?

Reply:
No, because that's a different metadata key. 😆 

I've toyed many times with optimizing a multi-key metadata get/set, but never got around to implementing it. After all, these are really just fields inside the JSON object value for the primary row key `server`, and it's inefficient to fetch that JSON blob twice. It just never quite hit the threshold, and is unlikely to now.

Reply:
D*mn.  I stared at that, too, trying to figure out why you'd coded two calls....  😊